libs/pando/engine/src/optimization/prune.ts

```
@@ -0,0 +1,566 @@
+      case 'thres':
+        r = { min: Math.min(...mins), max: Math.max(...maxs) }
+        break
+      case 'sumfrac':
+        if (mins[0] + mins[1] > 0 || maxs[0] + maxs[1] < 0)
+          r = cornerRange(n.op, ranges as [Range, Range])
+        // Degenerate if `x + c` touches zero
+        else r = { min: NaN, max: NaN }
+        break
+      case 'read': {
+        r = { min: 0, max: 0 }
+        for (const { [n.tag[cat]!]: range } of compRanges)
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)